pkg/operator/ceph/object/s3-handlers.go

```diff
@@ -199,12 +199,18 @@ func (s *S3Agent) DeleteObjectInBucket(bucketname string, key string) (bool, err
 
 func BuildTransportTLS(tlsCert []byte, insecure bool) *http.Transport {
 	//nolint:gosec // is enabled only for testing
-	tlsConfig := &tls.Config{MinVersion: tls.VersionTLS12, InsecureSkipVerify: insecure}
+	tlsConfig := &tls.Config{InsecureSkipVerify: insecure}
+	var caCertPool *x509.CertPool
+	var err error
+	caCertPool, err = x509.SystemCertPool()
+	if err != nil {
+		logger.Warningf("failed to load system cert pool; continuing without loading system certs")
+		caCertPool = x509.NewCertPool() // start with empty cert pool instead
+	}
 	if len(tlsCert) > 0 {
-		caCertPool := x509.NewCertPool()
 		caCertPool.AppendCertsFromPEM(tlsCert)
-		tlsConfig.RootCAs = caCertPool
 	}
+	tlsConfig.RootCAs = caCertPool
```

Comment:
will this work without appending the certs?

Reply:
We still need to append the user-selected cert in case the user uses self-signed certs.

The challenging part of understanding all this is that this code has to handle both internal and external cases. 

The internal case is easy, and there should be no need to use system certs. For internal object stores, the leaf certificate is what users should give. The leaf cert will be mounted by the RGW (server) that Rook configures, and Rook (as a client) can always use that to authenticate the RGW. The previous code Rook used implicitly assumed internal stores only, I think.

For external stores, the RGW (server) is not controlled by Rook, and Rook (as a client) can use the leaf cert, the intermediate cert that signed the RGW cert, or a CA to validate the server. In the external case, I've learned that users are best off giving the intermediate signing cert to the external script for self-signed certs. For well-known CAs (like letsencrypt), Rook can help out further by loading the system certs. Also of note, Rook can reliably expect system certs to be available because the system certs RPM package is a required dependency of the ceph-rgw RPM package.